R/markdown.R

```diff
@@ -256,6 +256,7 @@ mdxml_code_block <- function(xml, state) {
   info <- xml_attr(xml, "info")[1]
   if (is.na(info) || nchar(info[1]) == 0) info <- NA_character_
   paste0(
+    "\n\n",
```

Comment:
Doesn't this add an extra empty line? AFAIR in Rd you should not have an empty line or even a line break before `\preformatted{}` because that adds an empty line.

Why is it needed, btw?

Reply:
This makes `\preformatted` start its own paragraph (which doesn't get wrapped in `<p>`), otherwise you it starts before the text paragraph ends and you get `<pre></p>` in the rendered HTML.